ReverseTicTacToe/UI/ConsoleUIManager.cs

```diff
@@ -0,0 +1,265 @@
+                        {
+                            turnInputIsValid = v_IsValid;
+                            o_Row = row;
+                            o_Col = col;
+                        }
+                    }
+                }
+            }
+
+            return turnInputIsValid;
+        }
+        private static string removeWhiteSpaces(string i_UserTurnInput)
+        {
+            StringBuilder stringBuilder = new StringBuilder(i_UserTurnInput);
+            int index = stringBuilder.ToString().IndexOf(' ');
+            while (index != -1)
+            {
+                stringBuilder.Remove(index, 1);
+                index = stringBuilder.ToString().IndexOf(' ');
+            }
+            return stringBuilder.ToString();
+        }
+        private int getBoardSizeFromUser()
+        {
+            string userInputStr;
+            int boardSize;
+            StringBuilder msg = new StringBuilder();
+            msg.AppendFormat("Please enter the board size (choose between: {0} - {1})", k_MinBoardSize, k_MaxBoardSize);
+            msg.Append(Environment.NewLine);
+            do
+            {
+                WriteLine(msg);
+                userInputStr = ReadLine();
+            }
+
+            while (!isValidInput(userInputStr, out boardSize, k_MinBoardSize, k_MaxBoardSize));
+
+            return boardSize;
+        }
+        /// <summary>
+        /// This function get a string and checks if it is a number in range of i_Min, i_Max
+        /// returns true if it in range, and also if true return the number in o_Number,
+        /// else if false o_Number will be -1
+        /// </summary>
+        public bool isValidInput(string i_UserInputStr, out int o_Number, int i_Min, int i_Max)
+        {
+            int boardSize;
+            const bool v_ValidBoardSizeInput = true;
+            bool isValidBoardSizeInput = !v_ValidBoardSizeInput;
+            o_Number = k_InitVal;
+            if(int.TryParse(i_UserInputStr, out boardSize))
+            {
+                if(isInRage(boardSize, i_Min, i_Max))
+                {
+                    o_Number = boardSize;
+                    isValidBoardSizeInput = v_ValidBoardSizeInput;
+                }
+            }
+
+            return isValidBoardSizeInput;
+        }
+        private static bool isInRage(int i_Num, int i_Min, int i_Max)
+        {
+            return i_Num >= i_Min && i_Num <= i_Max;
+        }
+        public Logic.ReverseTicTacToe.eGameMode getGameTypeFromUser()
+        {
+            string userInputStr;
+            int gameTypeNum;
+            StringBuilder msg = new StringBuilder();
+            msg.Append("Please enter the game type ('1' for PVP, '2' for PVC):");
+            msg.Append(Environment.NewLine);
+            do
+            {
+                WriteLine(msg);
+                userInputStr = ReadLine();
+            }
+
+            while (!isValidInput(userInputStr, out gameTypeNum, k_MinGameType, k_MaxGameType));
```

Comment:
In this case is think it's better not to call "isValidInput" but to check manually  -
if(userInput != (int)eGameMode.PvP && userInput != (int)eGameMode.PvC) ..........
.
i think that would lessen the UI dependency in the Logic. If the Logic will decide in the future to change the gameMode numbers, it wouldn't have any effect on the UI

Reply:
but I don't have userInput as int, but as a string so I still have to Parse it to int, check if the parse succeeded and I make sure the number is in range of the possible gametypes count. so if in the future a new gameType will be added I won't need to change the code, whereas manually I do will need to change the code and add another statement to the if condition.
moreover, I don't see why should the number assigned to the enum change.